DEVEL.md

```diff
@@ -82,3 +82,30 @@ line more difficult.
 The default is to not log `Info()` calls. This can be changed using the plugin command
 line `-v` parameter. The additional annotations prepended to log lines by 'klog' can be disabled
 with the `-skip_headers` option.
+
+Error Conventions
+-----------------
+
+The framework has a convention for producing and logging errors. Ideally plugins will also adhere
+to the convention.
+
+Errors generated within or passed out of 'our' functions are
+[wrapped with their stack dump](https://godoc.org/github.com/pkg/errors#WithStack)
+with code such as:
+
+```golang
+    return errors.WithStack(err)
+```
+
+These errors are then logged using a [default struct value format](https://golang.org/pkg/fmt/#hdr-Printing)
+like:
+
+```golang
+    klog.Errorf("Example of an internal error death: %+v", err)
+```
+
+Errors generated from outside the plugins and framework can be logged as simple values:
+
+```golang
+    klog.Warningf("Example of a warning due to an external error: %v", err)
+```
```

Comment:
How about this variant? Does it make sense?

Errors generated within the framework and plugins are instantiated with the `New()` and `Errorf()` functions of the [`errors`](https://godoc.org/github.com/pkg/errors) package:

```golang
    return errors.New("error message")
```

Errors generated from outside the plugins and framework are augmented with their stack dump with code such as
```golang
    return errors.WithStack(err)
```
or
```golang
    return errors.Wrap(err, "some additional error message")
```

These errors are then logged using a [default struct value format](https://golang.org/pkg/fmt/#hdr-Printing) like:

```golang
    klog.Errorf("Example of an internal error death: %+v", err)
```

at the line where it's certain that the error cannot be passed out farther nor handled gracefully. Otherwise it's ok log them as simple values:

```golang
    klog.Warningf("Example of a warning due to an external error: %v", err)
```

Reply:
ack @rojkov - amended and pushed.